ThirdPartyAdapters/unity/unity/src/main/java/com/google/ads/mediation/unity/UnityEventSender.java

```diff
@@ -0,0 +1,57 @@
+package com.google.ads.mediation.unity;
+
+import android.util.Log;
+
+import com.google.ads.mediation.unity.eventlisteners.IUnityEventListener;
+
+public class UnityEventSender {
```

Comment:
I think this whole class could be omited basically + AdEvent as well removed
Since you created a common Interface for all listeners (maybe they should be called "adapters" since it adapts our callbacks to AdMob callback signatures), we can basically hold the listener themselves in the UnityBannerAd, UnityRewardedAd and UnityAdapter directly removing the need for UnityEventSender and the AdEvent enum itself.

Great work though! Let me know if you need more explanation we can go over it on a call basically :-)

Reply:
I see how I overcomplicated it :) Updating it now